src/main/java/core/basesyntax/service/impl/PurchaseAction.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ActionHandler;
+
+public class PurchaseAction implements ActionHandler {
+    public void apply(String fruit, int quantity) {
+        Storage.removeFruit(fruit, quantity);
+    }
```

Comment:
According to the checklist, the `PurchaseAction` class should check the result balance before saving it in the Storage - it should be positive. Throw a `RuntimeException` in case the balance is negative. Currently, the `apply` method directly removes the fruit from the storage without any balance checks.